File: app-geekhome/src/main/java/eu/geekhome/rest/configurable/FieldDefinitionDtoMapper.java
package eu.geekhome.rest.configurable;

import eu.geekhome.services.configurable.FieldDefinition;
import eu.geekhome.services.configurable.FieldDto;

public class FieldDefinitionDtoMapper {

    public FieldDto map(FieldDefinition<?> field) {
        return new FieldDto(field.getClass().getSimpleName() , field.getName(), field.getHint(),
                field.getMaxSize());
    }
}

File: app-geekhome/src/main/java/eu/geekhome/rest/configurable/FieldDefinitionDtoMapper.kt
package eu.geekhome.rest.configurable

import eu.geekhome.services.configurable.FieldDefinition
import eu.geekhome.services.configurable.FieldDto

class FieldDefinitionDtoMapper {
    fun map(field: FieldDefinition<*>): FieldDto {
        return FieldDto(
            field.javaClass.simpleName, field.name, field.hint,
            field.maxSize
        )
    }
}